zkevm-circuits/src/evm_circuit/execution.rs

```diff
@@ -1104,7 +1106,7 @@ impl<F: Field> ExecutionConfig<F> {
                             .map(move |step| (tx, &tx.calls()[step.call_index], step))
                     }))
                     // this dummy step is just for real step assignment proceed to `second last`
-                    .chain(std::iter::once((&dummy_tx, &last_call, &block.end_block)))
+                    .chain(std::iter::once((&dummy_tx, &last_call, padding)))
```

Comment:
Here dummy `next step` need to be padding, because padding assignment is idempotent, while `end_chunk`/`end_block` are not.
The reason is because `end_chunk`/`end_chunk` assignment function got `region.constrain_constant` which is un-revertable.

Here we use padding, later in `// part4: re-assigned real second last step, because we know next_step_after_real_step now` the correct next step will be used instead